Implement SecNum for monster stats damage/HP

AMXBus opened the pull request:
Another SecNum change but these are a bit more complex.

Main things to note:

- We implement SecNum calls in the runtime intit structure just like before for the monster damage/HP stats (currently in CREATURELOCKER.as and RebalancedCreatures.as if we ever want to use those)
- The calls are a little bit different here as we incorporate the champion monsters as well (these stats are in CHAMPIONCAGE.as)
- This is the first moment where I have seen pointers (primitive but it's something) being used in this game as CHAMPIONCAGE.as and CREATURELOCKER.as have their own getter functions for properties. (CHAMPIONCAGE.GetGuardianProperty() and CREATURES.GetProperty())
- All POPUP equivalents use these property getters that are essentially full object passers so we need to be careful if we plan any enhancements further down the line.
- Some buildings use these monster properties for their own functions like bunkers and champion cage healing (housing as well if MR3 is active)
- There is a lot of complex math that gets involved at every tick() but is well secured with prior SecNum implementation
- This should not interfere with how the game calculates values for buffs/armor/dmg reduction as all that math is well in the background and works on each tick(). If any changes would happen mid calculation it would either crash the game or correct itself on the next one as these values get recalculated... all the time...

React1-X commented on client/scripts/CHAMPIONCAGE.as:
```diff
@@ -480,6 +480,11 @@ package
          {
             var _loc2_:Object = _guardians["G" + param1];
             var _loc3_:Object = _loc2_.props;
+            for (var temp = 0; temp < _loc3_.health.length; temp++)
```
Hey @AMXBus - would you mind explaining the reason for introducing this loop here? I noticed it's also called "temp", is this accidentally included in the PR?

tram98 committed:
Web dockerfile: docker-start-web.sh was moved, so the path needs to be updated.
npm install changed to clean-install to respect package-lock

tram98 committed:
dockerignore: also exclude .env file. use environment from docker-compose instead

tram98 committed:
migrated docker-compose environment variables to global .env file. also added redis to docker-compose.yml

tram98 committed:
updated readme, changed node version to 18 to be in sync with prod. Improved documentation of environment variables in docker-example.env

tram98 committed:
Added ENV variable

React1-X committed:
Merge pull request #132 from tram98/docker-fixes

Improvements to Docker Compose setup

React1-X committed:
Update README.md

React-1 committed:
Minor cleanup on Docker config files

React-1 committed:
Removed TODO comments from infernoSave.ts

React-1 committed:
Initial setup for Inferno MapRoom - getNeighbours route & controller | Added dummy MapRoom users

React-1 committed:
Updated example.env - removed unused vars | Updated dicebear user profile pictures

tram98 committed:
Fixed an issue in docker-example.env. The http prefix must be present in BASE_URL, otherwise the client cannot make more web requests when requesting the maproom URLs

React-1 committed:
Damaged protection is now applied on takeover of wild monster cells

React1-X committed:
Merge pull request #133 from bym-refitted/docker-example-fix

Fixed an issue in docker-example.env. The http prefix must be present…

React-1 committed:
Tribes are now deterministic based on level ranges

React1-X committed:
Merge pull request #138 from bym-refitted/feature/tribe_levels

Tribes are now deterministic based on level ranges

React-1 committed:
Outpost cells now regenerate to 0% damage after damage protection is removed

React-1 committed:
Renamed models maproom1 and incident_report for consistency

React-1 committed:
createdAt and lastupdateAt updated on outpost takeover

React-1 committed:
Merge branch 'main' of https://github.com/bym-refitted/backyard-monsters-refitted into feature/inferno_maproom

React-1 committed:
Added Inferno MapRoom background asset | Added TODO comments

React-1 committed:
Experimental fix: Outpost regular kits are in building phase when attacked for the 2nd time

React-1 committed:
Reverted experimental changes

React-1 committed:
Fixed outposts resetting after 2nd attack | buildingdata validation

React1-X committed:
Merge pull request #139 from bym-refitted/bugfix/outpost_reset

Outposts now correctly persist building data with server-side validation

React-1 committed:
Improved server-side validation in buildingDataHandler for modified values

React-1 committed:
Merge branch 'main' of https://github.com/bym-refitted/backyard-monsters-refitted into feature/inferno_maproom

React-1 committed:
Generate tribes in Inferno MR on infernoBaseLoad scaled based on user level

React-1 committed:
Added moloch tribe 214

React-1 committed:
Added moloch tribe 215 - 217

React-1 committed:
Added moloch tribe 218 - 220

React-1 committed:
Implemented scaling moloch tribes | 7 tribes per level category

React-1 committed:
Inferno save: retrieve save by basesaveid

React-1 committed:
OneToOne relationship between user and infernoSave | infernoSave improvements

React-1 committed:
Improvements to infernoSave & zod schema validation

React-1 committed:
Moved handling of scaled MR1/Inferno tribes into services module

React-1 committed:
Removed redundant check for save in infernoSave.ts

React-1 committed:
Tribe respawn logic (2 hours) | Fixes to infernoSave.ts during an attack

React-1 committed:
TribeData interface in model updated to include optional properties for destroyed states

React-1 committed:
Added balanced resources to Moloch tribes

React-1 committed:
tribes within the user's current level range are stored

React-1 committed:
Fixed infernosave foreign key naming convention | set tribe respawn to 1hr

React-1 committed:
Reverted DevSettings.ts

React1-X committed:
Merge pull request #141 from bym-refitted/feature/inferno_maproom

Map Room 1 / Inferno Map Room Implementation [Tribes]

React-1 committed:
Added flag to enable/disable register form

React-1 committed:
Added koa2 rate limiter to register route

React1-X committed:
Merge pull request #142 from bym-refitted/feature/koa2-rate-limit

Added rate limiter to register route

React-1 committed:
Updated magma output from molochTribes

sir-kick-me2 committed:
Bug Fixes: Unbuilt buildings and mushrooms.
 - Unbuilt buildings no longer can get saved in buildingData before they are placed. This caused the building with ID 0 to be overwritten.
 - Unbuilt buildings no longer contribute to a base's value before they are placed.
 - Picking up a mushroom no longer causes workers to pause working on a building if that building shared an ID with the picked mushroom.

sir-kick-me2 committed:
Bug Fix: Removed erroneous checks in CaluclateExpectedTownHallLevel function.
- Town Hall will no longer automatically upgrade to level 4 when a save has over 2m XP, due to checking an always-true comparison.
- Under Hall will no longer automatically upgrade from level 5 to 6 when the four quake towers are built or when a magma tower is upgraded to level 4.

React1-X committed:
Merge pull request #143 from sir-kick-me2/thBugFixes

Bugfix: Townhall Up/Downgrading, Saving Unplaced Buildings, Mushrooms Canceling Worker

React-1 committed:
Beta version v1.1.6 release

React-1 committed:
Enabled version management in prod

React-1 committed:
Commented-out RebuildTH() from BASE.as due to bad-practice behaviour

React-1 committed:
Basic email JWT token verification for local

React-1 committed:
Added meetsDiscordAgeCheck bypass for local verifyJwtToken

React1-X committed:
Mailbox (Rewrite) Implementation  (#147)

* Feature: Add Mailbox functionality to BYMR (#140)

* initiate mail client & server

* change table structure to message & thread

* change user id when get all threads to the id of other user and remove unused code

* remove unused, refactor duplicated functions and add new comments

* refactor findOrCreateThread to service, update logging and update thread index

* Cleanup of models: 'thread' and 'message'

* move message.model db call logic to service (#144)

* Added constraint to message length with zod and MikroORM

* Removed old migration files from database/migrations

* Improved code formatting for mailbox changes

* Refactor and cleanup of getMessageTargets | mailboxErr error code | Additional improvements

* Refactored getMessageThread | Removed createDictionary util (type-safety improvements) | Additional cleanup

* Refactored getMessageThreads - cleanup

* Refactored findOrCreateThread | Improvements

* Added basic profanity filter to messages

---------

Co-authored-by: senninr <181501909+senninr@users.noreply.github.com>
Co-authored-by: React <bymdestroyers@gmail.com>

React-1 committed:
Bumped client & server to v.1.1.7-beta

React-1 committed:
Merge branch 'main' of https://github.com/bym-refitted/backyard-monsters-refitted

React-1 committed:
Chaos Weapons are now used in battles and must be remade in the Chaos Lab

sir-kick-me2 committed:
Fixed the academy popup object using the incorrect amount of inferno monsters. (#149)

sir-kick-me2 committed:
Fix: Outpost Upgrade Pausing, Loading older bases (#148)

* Renaming ProcessC variables for improved readability

* Removed "last processed" time's hard limit of two days. Refactored BASE processing function.
- Removed hard limit of 3000 ticks, now only limited to building/champion tick limit.
- Removed loop which was processing only 1 tick at a time regardless of tickDelta.

* Increased "tick limit" of building objects from 2 days to 2 weeks. Removed unused variable.
Re-added "last processed" time limit for 60 days, instead of 2, as the longer a base goes without a save, the longer its load time is.

React-1 committed:
Reduced savetime (last known save time) to 30 days in BASE.as

React-1 committed:
Formatting of BASE.as

React-1 committed:
Reduced savetime (last known save time) to 30 days in BASE.as

React-1 committed:
Bumped client and server to v1.1.8-beta

React-1 committed:
Bumped client and server to v1.1.8-beta

React1-X committed:
Upgraded to nodeJS version 22.14.0 | Updated package.json packages (#152)

Co-authored-by: React <bymdestroyers@gmail.com>

React-1 committed:
@mikro-orm core and mariadb packages to 5.9.8 | Regenerated package-lock.json

React-1 committed:
enabled skipLibCheck in tsconfig.ts - headache

React-1 committed:
Added license to package.json | cleanup package.json

React-1 committed:
Reverted new Date() for createdAt and lastUpdatedAt in takeoverCell.ts

React-1 committed:
Fixed imports in app.routes.ts

React-1 committed:
Reduced strictness on register route with rate limiter

React-1 committed:
Rate limiter now returns graceful error messages

MatiasBais committed:
Fix Visuals Errors On  HATCHERYCCPOPUP.as (#151)

* Fix visual errors on  HATCHERYCCPOPUP.as

* Update HATCHERYCCPOPUP.as

Changed param1 to creatureID in MonsterInfoB

* fix comments about fuctions change and rewritten MonterInfoB in HatcheryPopUp to match the one in HatcheryCCPopUp since its the same one

fix comments about fuctions change and rewritten MonterInfoB in HatcheryPopUp to match the one in HatcheryCCPopUp since its the same one

fix comments about fuctions change and rewritten MonterInfoB in HatcheryPopUp to match the one in HatcheryCCPopUp since its the same one

* fix extra _

AMXBus committed:
Fix sabnox not doubleing damage to favorite building targets (#153)

Co-authored-by: Marti Ali <marti.ali@kips.com.mk>

AMXBus committed:
Fix broken continuity when fortifying buildings (#123)

React-1 committed:
Bumped client and server to v1.1.9-beta

sir-kick-me2 committed:
Optimizing BRESOURCES, Reducing Long Load Times Caused by Harvesters (#154)

* Adding a parameter to BRESOURCE's produce function to allow it to produce multiple times at once.

* Changed BRESOURCE new optimized tick function to work with non-repairing buildings instead of just full health ones.

React-1 committed:
Bumped client & server to v1.2.0-beta

React-1 committed:
Bumped client & server to v1.2.0-beta

sir-kick-me2 committed:
Fink, Bandito, and Eyera Restorations. Wormzors Bugfix (#155)

* Gave ignore building and ignore creep parameters to targeting's get targets function.
Gave monsters a flag to turn on/off automatically calculating rotation.

* Gave ignore initial target and inner damage radius for AoeDamage ability components.
Fixed targeting parameters for Fink and Bandito ability components.
Renamed variables and parameters.

* Implementation for monster's AOEdamage ability to ignore friendly buildings, ignore currently targeted monster, and use inner radius parameter.
Changed wrong import, and damage parameter of DealLinearAEDamage to use absolute value since it expects a positive number.
Fixed comparison that was making AoeDamageOnAttack component have a minimum recharge duration of 1s.

* Restructuring getAllTargets. Fixing eyera's targeting flags and explosion on death.

React-1 committed:
Reverted CreepBase.as change which caused unintended behaviour with Eye-ra

AMXBus committed:
Merge branch 'main' into SecNumMonsterStats

React-1 committed:
Added newmap_init_setup and newmap_discord_age to portuguese and spanish translations

React-1 committed:
Bumped client & server to v1.2.1-beta

React1-X committed:
Server-side validation improvements (#156)

* Created server endpoint for /base/attack | ATTACK.as simple URLLoaderApi request

* Improved request structure in ATTACK.as

* Monster and champion stats sent to baseLoad.ts attack mode endpoint

* Moved validateAttack.ts to service module

* Monster stats are now validated on the server

* Improved project structure | zod directory | cleanup of validateAttack.ts

* Additional clean-up fixes for validateAttack.ts

* Champion stats now validated on server

* URLLoaderApi.as - param changed to compile-time constant | Fixed ASDoc

* Removed powerlevel check | Improved isChampionStatsEqual logic

---------

Co-authored-by: React <bymdestroyers@gmail.com>

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats

React-1 committed:
Added takeoverDate for outposts in save table

React-1 committed:
Organised MapRoom v2 world gen files | crossdomain.xml changes

React-1 committed:
Updated crossdomain.xml

React-1 committed:
Moved generateMap.ts into services module

React1-X committed:
D.A.V.E Club integration (#158)

Co-authored-by: React <bymdestroyers@gmail.com>

React-1 committed:
Inferno monsters unlockable in overworld with D.A.V.E Club

React-1 committed:
Added back critical MR3 UI elements which got removed from SWF

React-1 committed:
Added default position for MR3 co-ordinates UI button bar

AMXBus committed:
Fix oversight where worker messes up while choosing between upgrade or fortify rescources when upgrading anything that does not have a fortify property set (#157)

Co-authored-by: Marti Ali <marti.ali@kips.com.mk>

sir-kick-me2 committed:
Bugfix: Infinite DAVE Statues, Decorations Pausing Workers (#159)

* Fix to DAVE statue rewarder awarding duplicate statues by adding "cls" value to the dave statue prop.
Removed extra Constructed() call from BDECORATION since Place() already calls Constructed().

* Adding cls value to DAVE statue in inferno and outpost yard props

React-1 committed:
Removed 'researchdata' persistence from mapUserSaveData.ts | v1.2.3-beta

React-1 committed:
Merge branch 'main' of https://github.com/bym-refitted/backyard-monsters-refitted

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats

React1-X committed:
Homebase now always points to the user's main yard save | outposts do not have homebases (#161)

Co-authored-by: React <bymdestroyers@gmail.com>

React1-X committed:
Leaderboards controller/route implementation (#162)

Co-authored-by: React <bymdestroyers@gmail.com>

React-1 committed:
Added composite index on worldid, type and userid for save table | removed rate limit on leaderboards

React-1 committed:
Removed pic_square from getLeaderboards.ts query

React-1 committed:
Joining a world on the Map Room is now randomized

React-1 committed:
stats added to mapUserSaveData.ts

React1-X committed:
Attack Logs API implementation  (#163)

* Added attacklog model | basic implementation for creating attack logs

* Added routes and controllers for attack logs implementation

* Implemented server-side query filter for attack logs (redis cache)

---------

Co-authored-by: React <bymdestroyers@gmail.com>

React-1 committed:
Added API versioning to resert-password route

React-1 committed:
DB Migration: added indexes to attack_logs table

React-1 committed:
Added monthly shiny script for DAVE Club

React-1 committed:
Updated monthly-shiny script

React-1 committed:
Updated JSDoc comments in monthly-shiny script for how-to-use

React-1 committed:
Updated JSDoc comments in monthly-shiny script for how-to-use

React-1 committed:
Added safeguard to monthly-shiny.ts

React-1 committed:
Migration file: monthly_credits added to save model

React-1 committed:
Removed hardcoded 'main' string in place of enum in monthly-shiny.ts

React-1 committed:
Inferno lockerdata is now always synchronized with the main user save

React-1 committed:
Drastic improvements to zoom feature on Android

React-1 committed:
Removed Android-specific login from main branch

React-1 committed:
Enabled Koa proxy | IP logged

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats

React-1 committed:
Beta v1.2.4 | GAMELOADER.as added

React-1 committed:
Removed GAMELOADER.as from main repo

sir-kick-me2 committed:
Monster Transfer Bugfixes: Bases Swap Monsters, Base Not Showing Monsters (#160)

* Server now orders the source base the first result when transfering monsters.
Fixes an issue where monsters would sometimes swap between bases after a transfer.

* Outposts now preserve data that was previously lost after a monster transfer.
Fixes the following issues:
- After a transfer, outposts couldn't transfer monsters again until the map was reloaded.
- When the map was reloaded, the outpost would appear to have no monsters until the

* Updating variable names for transfer monster functions for readability.

* Changing the source cell of a monster transfer to be a copy rather than a reference.
Changing the scope of the target cell of a monster transfer so it can be referenced anywhere in MapRoom.as
Making transfers skip logic of updating monster data locally if no monsters were transferred.
Adding a _monsterTransfer[src] > 0 check to fix a minor bug where transfers would show "all monsters tranferred" if the target cell's housing was full.

* Modifying TransferMonsters and RequestData logic to be able to wait for other server requests to finish before sending a request.
TransferMonsters now updates source/target cell's monster data locally within the s tored zone data after a successful transfer.
Added helper functions GetCellZone, ZoneHasPendingTransferRequest, GetPendingZoneRequestIndex.
Refactored GetCell to use the new GetCellZone helper function. Functionality unchanged.

* Changing variable name _monsterTarget to _monsterTargetRef for consistency.
Server now verifies that both bases have the same user save id before a monster transfer.

React-1 committed:
Client and server bumped to v.1.2.5-beta

React-1 committed:
monthly-shiny script early return instead of process.exit(0)

React-1 committed:
Bumped framerate to 40 FPS for BYMR

React1-X committed:
Migration to Postgres (#165)

Co-authored-by: React <bymdestroyers@gmail.com>

React1-X committed:
Updated mikro-orm.config.ts to define default Postgres schema on init

React1-X committed:
attackreport column in save table type change to text

React1-X committed:
Store 'champion' as JSONB - return stringified to client

React1-X committed:
Fixed updateSaved route | removed problematic championHandler.ts | Migration file

React1-X committed:
Added 'not currently in use' JSDoc comment for championHandler.ts

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats

React1-X committed:
Significant asset recovery dump

React1-X committed:
Bumped client and server to v1.2.6-beta | Reverted missing Fomor level 3 asset

React1-X committed:
Temporarily disabled base migration due to issues

React1-X committed:
Updated Docker configurations for Postgres

React1-X committed:
Docker only exposes ports on dev profile | default ports | pgadmin v9.4

React1-X committed:
Added docker-compose.prod.yml

React1-X committed:
Added multi-stage Dockerfile | Alphine 22.14.0

React1-X committed:
Merge pull request #166 from bym-refitted/fix/docker

🐳 Docker: Postgres implementation

React1-X committed:
Update README.md

React1-X committed:
Adds constraint check for 'credits' to be a non-negative value

React1-X committed:
Reduced King Wormzer and Balthazar academy levels for final descent base

React1-X committed:
Updated example.env

React1-X committed:
Base migration enabled: wrapped operation in transaction to ensure atomicity w/ validation

React1-X committed:
Reduced the difficulty level of Moloch tribes above levels 21 | Increase magma loot

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats

React1-X committed:
MapRoom fullscreen mode fix

React1-X committed:
Merge pull request #167 from bym-refitted/feature/mr2-resize

Fixed 'fullscreen' mode on Map Room 2

sir-kick-me2 committed:
Changing getMR2MonsterUpdateSaveData variable names for readability.

sir-kick-me2 committed:
During an attack, monsters are no longer removed from bases' monsterdata before being flung.
Making variable _creaturesLoaded properly decrement when a champion is deselected from being flung.

React1-X committed:
Updated example.env to include local postgres password

sir-kick-me2 committed:
Changing variable names in RezghulResurrectAttack for readability.

sir-kick-me2 committed:
Rezghul no longer uses their resurrect attack when not in an attacking/defending state.
Fix for a game freeze when Rezghul resurrected a defending monster.

React1-X committed:
Merge pull request #169 from sir-kick-me2/RezghulFreeze

Bugfix: Rezghul using ability outside of battle state. Resurrecting defending monsters freezes game.

React1-X committed:
Merge pull request #168 from sir-kick-me2/FlingerMonsters

Bugfix: Monsters removed from housing before being flung. Minor oversight when deselecting champion.

React1-X committed:
Bumped client and server to v1.2.8-beta

React1-X committed:
passwords no longer require a number

React1-X committed:
Moved specialevent & zazzle assest into correct directories

React1-X committed:
Added MR3 tribe portrait assets

React1-X committed:
Recovered quest assets

sir-kick-me2 committed:
Yard type is no longer set to Outpost prematurely when loading the player's next outpost.
Game now waits until the current base is no longer saving before loading the next base.

sir-kick-me2 committed:
King Wormzor's splash damage no longer hits monsters.
Refactored getAllTargets function in AOEDamage Component. Now uses MonsterBase instead of CreepBase to account for champions.

React1-X committed:
Merge pull request #170 from sir-kick-me2/MissingChampions

Bugfix: Champion Data Reset. AOEDamage Against Champions; King Wormzor Change.

React1-X committed:
Bumped client and server to v1.2.9-beta

React1-X committed:
logMissingAssets in DevSettings.ts disabled for prod

React1-X committed:
Update README.md

React1-X committed:
Update README.md

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats

React1-X committed:
Added view logic & testing getNeighbours route

React1-X committed:
.env creation moved to package.json as build-env.mjs script

React1-X committed:
Merge branch 'main' of https://github.com/bym-refitted/backyard-monsters-refitted

React1-X committed:
Moved build-env.mjs into src/scripts directory

React1-X committed:
Update CONTRIBUTING.md

React1-X committed:
Remove whitespace from email and password fields with zod - AuthSchema.ts

React1-X committed:
Merge branch 'main' of https://github.com/bym-refitted/backyard-monsters-refitted

React1-X committed:
Reverted regex refactor

React1-X committed:
Improved bone harvester placeholder asset/animation

React1-X committed:
Fixed local.swf through the launcher - loaderInfo.params correctly set in client

React1-X committed:
loaderInfo.params correctly set in client for GAME.as

React1-X committed:
merged origin main

React1-X committed:
Added 'getNeighbours' controller implementation for Inferno Map Room

React1-X committed:
Updated queries to fetch same-world users from MR2

React1-X committed:
Performance optimizations to getNeighbours - single query to fetch infernoSaves by worldid | restructured composite index

React1-X committed:
Removed single-line comments from getNeighbours.ts

React1-X committed:
Minor improvements to code quality in getNeighbours.ts

React1-X committed:
Inferno PvP attack mode, attack notifications & attack logs

React1-X committed:
Monsters die in battle for PvP inferno

React1-X committed:
Added all data properties the client expects for a neighbour

React1-X committed:
Removed lastupdateAt from data object - not expected by the client

sir-kick-me2 committed:
- Balthazar: Removed flying and ground defense flags; can now be targetted by anything that targets attackers.
- findHuntingTargets: Krallen is now counted as a target. Renamed variables.
- added _hasLivingGuardian function to CREATURES
- Defending Balthazars now deal *3 damage to attacking monsters.

marti-kips committed:
Minor fixes to some Inferno bugs

marti-kips committed:
Quick fix for Balthazar not being targetable by most buildings/monsters

sir-kick-me2 committed:
Replace broken while loop with for loop for _hasLivingGuardian function.

React1-X committed:
Adds attacker to defenders list of users when attacked

React1-X committed:
Added damage protection, syned-neighbours across attacks & retaliation mechanics

React1-X committed:
Additional clean-up across neighbour system

React1-X committed:
Updated migration files

React1-X committed:
Reverted DevSettings/ts changes

React1-X committed:
Set neighbors 'defaultRaw' to be empty array in infernomaproom.model.ts

React1-X committed:
Merge pull request #175 from bym-refitted/feature/inferno-maproom-pvp

Inferno PvP Implementation

React1-X committed:
getNeighbours edge-case: early return if user does not have a worldid - do not set cache

React1-X committed:
Changed neighbors from json to jsonb data type in infernomaproom.model.ts

React1-X committed:
Resources can now be looted in Inferno PvP

React1-X committed:
Merge pull request #174 from sir-kick-me2/Balthazar

Balthazar: Targeting Krallen, Defense Flags, Damage Multiplier when Defending.

React1-X committed:
Bumped client and server to beta v1.3.1

React1-X committed:
Bumped client and server to beta v1.3.1

React1-X committed:
Fixed duplicate main yards due to logical flaw in leaveWorld.ts

marti-kips committed:
Revert "Quick fix for Balthazar not being targetable by most buildings/monsters"

Since it's fixed.

AMXBus committed:
Merge branch 'bym-refitted:main' into InfernoFixes

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats

React1-X committed:
Fixed Inferno hatchery speed-up timers not resetting

React1-X committed:
Merge pull request #173 from AMXBus/InfernoFixes

Fixes to minor Inferno bugs

React1-X committed:
Reverted removing DAVE Club resourceBar from UI_TOP.as

React1-X committed:
Bumped client & server to beta v.1.3.2

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats

React1-X committed:
Added several WMI assets, Wiki assets & archived assets

React1-X committed:
Reverted names for wild monster invasion 1 asseets to correct file names

React1-X committed:
Added original Wild Monster Invasion 1 files back into codebase

This commit includes the logic from the original WMI 1 event - these files where rewritten by the original developers instead of creating a new class, they used existing logic and updated it for WMI 2, this commit aims to introduce the original logic by creating files suffixed with _WM1

React1-X committed:
Updated SPECIALEVENT_WM1.as to include correct Tick() function before removal

React1-X committed:
Updated WMI 1 logic with original event logic before it was hardcode disabled

React1-X committed:
Added additional WMI 1 logic present from game-v120.v7

React1-X committed:
EventActive() returns true

React1-X committed:
Fixed bugs with old implementation (._mode and ._isOutpost)

React1-X committed:
Added 200x200_1.jpg popup asset

React1-X committed:
Fixed UI_NEXTWAE_WM1.ShouldDisplay()

React1-X committed:
Functioning WMI1 with some bugs

React1-X committed:
Monsters now retreat when surrender occurs

React1-X committed:
Updated comments in SPECIALEVENT_WM1.as

React1-X committed:
Fixes monsters retreating during waves

React1-X committed:
Fixed champions spawning & added in originl logical checks

React1-X committed:
Included bonus wave 2

React1-X committed:
Totem levels are now tracked

React1-X committed:
Added basic server-side timers for Wild Monster Invasion event

React1-X committed:
Disabled WMI for Inferno & basic server-side event handling

React1-X committed:
Improved naming conventions

React1-X committed:
Improvements to event phases on the server

React1-X committed:
Final improvements for server-side logic for event

React1-X committed:
Fixed multiple totems being placed down

React1-X committed:
Bumped client & server to v1.3.3-beta

React1-X committed:
Merge pull request #177 from bym-refitted/feature/wild-monster-invasion

[WIP] Add Wild Monster Invasion 1 to BYMR

React1-X committed:
Fixed BFOUNDATION logic from WMI event causing issues on outposts

React1-X committed:
Performance optimizations for client: Monster spawning, targetting & movement

React1-X committed:
Added gold totem building asset

React1-X committed:
YARD_PROPS.as - now shows golden totem buildingbutton

AMXBus committed:
Merge branch 'bym-refitted:main' into SecNumMonsterStats
